content/docs/1.7.0/nodes-and-volumes/volumes/trim-filesystem.md

```diff
@@ -54,23 +58,53 @@ You can set up a [RecurringJob](../../../snapshots-and-backups/scheduling-backup
+To help reclaim as much space as possible automatically, Longhorn includes the setting
+[_Remove Snapshots During Filesystem Trim_](../../../references/settings/#remove-snapshots-during-filesystem-trim).
+This allows the trim feature to automatically mark the latest snapshot and all preceding snapshots
+(until there is a fork in the snapshot chain) as removed. As a result, Longhorn can reclaim space for as many snapshots
+as possible. However, this setting also causes intentionally created snapshots to be marked as removed (and eventually
+purged), so it should be used with caution.
 
-#### The Volume Spec Field "UnmapMarkSnapChainAsRemoved"
+### The Volume Spec Field "UnmapMarkSnapChainAsRemoved"
 
-Of course there is a per-volume field `volume.Spec.UnmapMarkSnapChainAsRemoved` would overwrite the global setting mentioned above.
+There is a per-volume field `volume.Spec.UnmapMarkSnapChainAsRemoved` that overwrites the global setting mentioned above.
 
-There are 3 options for this volume field: `ignored`, `enabled`, and `disabled`. `ignored` means following the global setting, which is the default value.
+There are 3 options for this volume field: `ignored`, `enabled`, and `disabled`. `ignored` means followi the global
+setting, which is the default value.
 
-You can directly set this field in the StoragaClasses so that the customized value can be applied to all volumes created by the StorageClasses.
+You can directly set this field in a StoragaClass so that the customized value can be applied to all volumes created by
+the StorageClass.
 
 ## Known Issues & Limitations
 
+### Rebuilding volumes
+
+The trim operation cannot be executed successfully against a volume that is actively rebuilding. By design, it unmaps
+blocks in the volume head and all the continuously removed snapshots preceding it. However, some of these snapshots may
+be actively transferring from one replica to another while rebuilding is ongoing.
+
+Instead of returning an I/O error to the filesystem, Longhorn silently refuses to unmap blocks during a rebuild. The
+rebuild may take a long time, and VM workloads in particular respond poorly when repeated attempts to complete a trim
+return errors. See https://github.com/longhorn/longhorn/issues/7103 for details.
+
+If a trim operation is started during a rebuild, it will have no effect. Similar to issuing a trim against a valid
+snapshot, the filesystem will remember that it has already trimmed the associated blocks. Later on, if you want to
+attempt the trim again (to recover space), you may need to unmount and remount the filesystem first.
+
+### Expanding volumes
+
+The trim operation cannot be executed successfully against a volume that is actively expanding. Unlike for rebuilding
+volumes, Longhorn returns an I/O error to the filesystem when this occurs. Expansion is generally fast. It is better
+to inform the filesystem that the trim failed so that it can be tried again soon (without remounting).
```

Comment:
```suggestion
Longhorn is unable to trim the filesystem during volume expansion and returns an I/O error whenever the issue is encountered. Consider informing the filesystem about any unsuccessful trim operation so that another attempt can be made without the need for remounting.
```

Reply:
This one also changes the intended meaning. Please check my new wording.